geo/src/algorithm/concavehull.rs

```diff
@@ -0,0 +1,383 @@
+use crate::algorithm::convexhull::ConvexHull;
+use crate::algorithm::euclidean_distance::EuclideanDistance;
+use crate::algorithm::euclidean_length::EuclideanLength;
+use crate::prelude::Centroid;
+use crate::utils::partial_min;
+use crate::{Line, LineString, MultiLineString, MultiPoint, MultiPolygon, Point, Polygon};
+use num_traits::Float;
+use rstar::{RTree, RTreeNum};
+use std::collections::VecDeque;
+
+fn find_point_closest_to_line<T>(
+    interior_points_tree: RTree<Point<T>>,
+    line: Line<T>,
+    max_dist: T,
+) -> Option<Point<T>>
+where
+    T: Float + RTreeNum,
+{
+    let w = line.euclidean_length() + max_dist.clone() + max_dist.clone();
+    let h = max_dist.clone() + max_dist.clone();
+    let two = T::add(T::one(), T::one());
+    let search_dist = T::div(T::sqrt(T::powi(w, 2) + T::powi(h, 2)), two);
+    let mut candidates = interior_points_tree
+        .locate_within_distance(line.centroid(), search_dist)
+        .peekable();
+    let peek = candidates.peek();
```

Comment:
would the rest of this method be cleaner implemented using `candidates.min_by(...`?

Maybe not once you handle `NaN`... so take it or leave it.